From e26cf1812907c31c51be00d0535bc29786f17dba Mon Sep 17 00:00:00 2001
From: Roberto Sartori <roberto.sartori.android@gmail.com>
Date: Sun, 25 Jun 2023 14:24:58 +0200
Subject: [PATCH] netutils_wrapper/network_stack: allow netutils_wrapper to
 write fs_bpf_netd_shared

Needed (at least) with OnePlus blobs from 4.4.

Change-Id: If484a2b20587fda45de2a5b761a6596f734ad573
Signed-off-by: Roberto Sartori <roberto.sartori.android@gmail.com>
---
 private/network_stack.te | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/private/network_stack.te b/private/network_stack.te
index d9135a1ea..840b62a3b 100644
--- a/private/network_stack.te
+++ b/private/network_stack.te
@@ -71,6 +71,9 @@ allow network_stack self:netlink_xfrm_socket { create_socket_perms_no_ioctl nlms
 allow network_stack tun_device:chr_file rw_file_perms;
 allowxperm network_stack tun_device:chr_file ioctl { TUNGETIFF TUNSETIFF TUNSETLINK TUNSETCARRIER };
 
+# Used with old vendor blobs from OnePlus
+allow netutils_wrapper fs_bpf_netd_shared:file write;
+
 ############### NEVER ALLOW RULES
 # This place is as good as any for these rules,
 # and it is probably the most appropriate because
@@ -94,7 +97,6 @@ neverallow netd fs_bpf_netd_readonly:file write;
 # netutils_wrapper requires access to be able to run iptables and only needs readonly access
 neverallow { domain -bpfloader -netd -netutils_wrapper -network_stack -system_server } fs_bpf_netd_shared:dir ~getattr;
 neverallow { domain -bpfloader -netd -netutils_wrapper -network_stack -system_server } fs_bpf_netd_shared:file *;
-neverallow netutils_wrapper fs_bpf_netd_shared:file write;
 
 # S+: Only the bpfloader and the network_stack should ever touch 'fs_bpf_tethering' programs/maps.
 neverallow { domain -bpfloader -network_stack } fs_bpf_tethering:dir ~getattr;
-- 
2.34.1

